### PR TITLE
Add Kernel Version Check

### PR DESCRIPTION
**Pull request by HTRamsey**
Adds a check for minimum supported kernel version

### CODE REVIEW

File: source/include/FreeRTOSIPConfigDefaults.h [hunk 1]
```diff
@@ -148,6 +148,10 @@
 
 /*---------------------------------------------------------------------------*/
 
+#if ( ( tskKERNEL_VERSION_MAJOR < 8 ) || ( ( tskKERNEL_VERSION_MAJOR == 8 ) && ( tskKERNEL_VERSION_MINOR < 1 ) ) )
```
Comment (AniruddhaKanhere) [inline source/include/FreeRTOSIPConfigDefaults.h:151]: I suppose that this specific version is used due to the fact that before this `pdMS_TO_TICKS` did not exist which is used a lot in this library.

**Review by AniruddhaKanhere — APPROVED** [review]
The changes look good to me.

Thank you @HTRamsey for taking the time to contribute!

Thanks

File: source/include/FreeRTOSIPConfigDefaults.h (hunk 1 above)
Comment (HTRamsey) [reply source/include/FreeRTOSIPConfigDefaults.h:151]: It's actually from [this](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1084#issuecomment-1997748807) but looking at the changelog myself I think maybe it should be v9.0 cause that's when task notifications were added. I can try and at least see the oldest version that will compile later.

Comment (AniruddhaKanhere) [reply source/include/FreeRTOSIPConfigDefaults.h:151]: Thank you! That would be great.

**Comment by AniruddhaKanhere** [comment]
@ActoryOu should we be including the kernel headers here https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/df5aed9e58a72e2489b6c1936788885672c62981/test/build-combination/Header_Self_Contain/headerSelfContain.cmake#L21 before compile options as the build check is failing there.

File: source/include/FreeRTOSIPConfigDefaults.h (hunk 1 above)
Comment (tony-josi-aws) [reply source/include/FreeRTOSIPConfigDefaults.h:151]: @HTRamsey 

Task notifications are used by the IP stack only if `ipconfigSELECT_USES_NOTIFY` is enabled. Network interfaces might still use it, depending on how the EMAC task is implemented.

**Comment by ActoryOu** [comment]
> @ActoryOu should we be including the kernel headers here
> 
> https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/df5aed9e58a72e2489b6c1936788885672c62981/test/build-combination/Header_Self_Contain/headerSelfContain.cmake#L21
> 
> before compile options as the build check is failing there.

Sorry for late response.
Yeah, that helps. And @HTRamsey created a better solution by #1120.
Let's do it that way.

**Comment by HTRamsey** [comment]
@tony-josi-aws to your first point, the value would be in making it obvious to users why it is not compiling, instead of getting more questions on the forum. To the second point, the check is currently for v8.1, not v8.2.

**Pull request closed without merging**